lib/usb/elgato-streamdeck.js

```diff
@@ -0,0 +1,216 @@
+/*
+ * This file is part of the Companion project
+ * Copyright (c) 2018 Bitfocus AS
+ * Authors: William Viker <william@bitfocus.io>, Håkon Nessjøen <haakon@bitfocus.io>
+ *
+ * This program is free software.
+ * You should have received a copy of the MIT licence as well as the Bitfocus
+ * Individual Contributor License Agreement for companion along with
+ * this program.
+ *
+ * You can be released from the requirements of the license by purchasing
+ * a commercial license. Buying such a license is mandatory as soon as you
+ * develop commercial activities involving the Companion software without
+ * disclosing the source code of your own applications.
+ *
+ */
+
+var openStreamDeck = require('@elgato-stream-deck/node').openStreamDeck
+var util = require('util')
+var common = require('./common')
+var debug = require('debug')('lib/usb/elgato_streamdeck')
+var sharp = require('sharp')
+var image_write_queue = require('./image-write-queue')
+var setTimeoutPromise = util.promisify(setTimeout)
+
+function elgato_streamdeck(system, devicepath) {
+	var self = this
+
+	self.config = {
+		brightness: 100,
+		rotation: 0,
+		page: 1,
+	}
+
+	process.on('uncaughtException', function (err) {
```

Comment:
shouldn't we just be doing this upstream in the thread handler?

Reply:
yeah probably, but it was already here so I didnt question it